pandas/core/dtypes/dtypes.py

```diff
@@ -86,7 +86,7 @@ class PandasExtensionDtype(ExtensionDtype):
     base: Optional[DtypeObj] = None
     isbuiltin = 0
     isnative = 0
-    _cache: Dict[str_type, PandasExtensionDtype] = {}
+    _cache_dtypes: Dict[str_type, PandasExtensionDtype] = {}
```

Comment:
the changes in this file and in pandas/tests/dtypes/test_dtypes.py make sense, the rest i think are unnecessary and in the case of cache_readonly harmful